docs/design/bundle-sources.md

```diff
@@ -0,0 +1,300 @@
+    ```
+
+1. Make the following changes to your Dockerfile:
+
+    *Example Dockerfile*
+
+    ```sh
+    FROM scratch
+    COPY manifests /manifests
+    ```
+
+    > **Note:** Use the `FROM scratch` directive to make the size of the image smaller.
+
+1. Add the following labels to your Dockerfile:
+
+    ```sh
+    LABEL operators.operatorframework.io.bundle.mediatype.v1=plain+v0
+    LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
+    LABEL operators.operatorframework.io.bundle.package.v1=<operator_name>
+    LABEL operators.operatorframework.io.bundle.channels.v1=<channels>
+    LABEL operators.operatorframework.io.bundle.channel.default.v1=<default_channel>
+    ```
+
+1. Build an OCI-compliant image using your preferred build tool, similar to the following example. You must use an image tag that references a repository where you have push access privileges.
+
+    *Example build command*
+
+    ```sh
+    docker build -f Dockerfile.plainbundle -t \
+    quay.io/<image_repo>/plainbundle:example .
+    ```
+
+1. Push the image to your remote registry:
+
+    ```sh
+    docker push quay.io/<image_repo>/plainbundle:example
+    ```
+
+### Additional resources
+
+* [File-based catalog bundle schema](https://github.com/operator-framework/olm-docs/blob/master/content/en/docs/Reference/file-based-catalogs.md)
+* [OCI image specification](https://github.com/opencontainers/image-spec#oci-image-format-specification)
+* [RukPak > Building a plain bundle > Image source](https://github.com/operator-framework/rukpak/blob/main/docs/bundles/plain.md#image-source)
+* [RukPak > Sources > Images > Private image registries](https://github.com/operator-framework/rukpak/blob/main/docs/sources/image.md#private-image-registries)
+
+## Creating a file-based catalog
+
+If you do not have a file-based catalog, you must perform the following steps to initialize the catalog.
+
+<h3 id="prereqs-creating-a-fbc">
+Prerequisites
+</h3>
+
+* `opm` CLI tool
+* Docker or Podman
+
+<h3 id="proc-creating-a-fbc">
+Procedure
+</h3>
+
+1. Create a directory for the catalog by running the following command:
+
+    ```sh
+    mkdir <catalog_dir>
+    ```
+
+1. Create a Dockerfile that can build a catalog image by running the `opm generate dockerfile` command:
+
+    *Example `<catalog_name>-image.Dockerfile`*
+
+    ```sh
+    opm generate dockerfile <catalog_dir>
+    ```
+
+    > **Note:** If do not want to use the default upstream base image, you can specify a different image using the `-i` flag.
+
+    * The Dockerfile must be in the same parent directory as the catalog directory that you created in the previous step:
+        *Example directory structure*
+
+        ```sh
+        .
+        ├── <catalog_dir>
+        └── <catalog_dir>.Dockerfile
+        ```
+
+1. Populate the catalog with the package definition for your Operator by running the `opm init` command:
+
+    ```sh
+    opm init <operator_name> \
+    --default-channel=preview \
```

Comment:
Is this flag required on the `opm init` command? If not, I'd suggest removing it because default channel is not an OLMv1 concept.

If it is required, should we consider an alternate instruction that has the user use HEREDOC to `cat` and `olm.package` directly into a file?

My motivation here is to avoid leaking OLMv0-specific APIs/concepts/behaviors into OLMv1 stuff.

Reply:
That makes sense. So, the flag itself is not required and I've removed it from the command. However, without the flag too, `opm init` generates an empty default channel in the index `"defaultChannel": ""`. 

I am guessing we should remove this line manually from the index. I _think_ it should be ok for this doc to use the `opm init` and then explicitly mention about removing the empty default channel from the index file. Open to knowing about any other alternatives.